### ISSUE TITLE
split and multi-shot senders

### ISSUE THREAD

**Issue by msimberg** [NVIDIA/stdexec#238]
4.7 says that senders can be single-shot or multi-shot. The former can only be connected once. The latter multiple times.

4.20.10 says that `split` will simply return the provided sender if it's a multi-shot sender, and otherwise adapt the provided sender into a multi-shot sender.

9.6.5.13 mentions nothing about multi-shot senders.

My expectation of `split` is that a graph of work such as the following would print `hello` only once:
```
auto s = split(then(just(), []() { cout << "hello\n"; }));
sync_wait(s);
sync_wait(std::move(s));
```
However, I'd also expect the sender returned from `then(just(), []() { ... })` to be copiable, and thus a multi-shot sender. If it is a multi-shot sender the example would be equivalent to:
```
auto s = then(just(), []() { cout << "hello\n"; });
sync_wait(s);
sync_wait(std::move(s));
```
according to 4.20.10, and `hello` would be printed twice.

One of these expectations must be wrong. Which one? Is 4.10.10 just outdated?

In HPX I'm currently treating `split` and `ensure_started` as the lazy and eager version of the same thing. `ensure_started` and `split` on a sender returned from `ensure_started` simply returns the sender unchanged. `split` on a sender returned from `split` returns the sender unchanged.

**Comment by kirkshoop** [comment]
In RX this is the difference between 'hot' and 'cold'. 

just() is 'cold'. every time it is started it produces the same result. 

mouse_click() is 'hot'. every time it is started it will only produce the next click (a different click)

Multi-shot (aka multicast in RX) is independent of 'hot' and 'cold'. Something 'cold' and multi-shot will repeat the operation each time that it is started. Something 'hot' and multi-cast will produce a different result each time it is started. 

The act of sharing a single result is independent of 'hot' and 'cold'
Sharing a result requires storage across different operations, 'hot' and 'cold' do not. Storage usually must be allocated. Rx calls the shared algo "replay". Replay is used to convert a 'hot' source into a 'cold' source

Rx also can convert a 'cold' source into a 'hot' source. That algo is called "publish". Publish has no storage and only starts the source once. To receive the result, the receiver must be attached when the result is emitted. Attaching after will never complete with the result. 

IMO

split should not try to detect multi-cast. It is precisely because of side-effects like printf() that this matters. split should always have the semantics of "replay" - only run the source once and store the result to share.

**Comment by kirkshoop** [comment]
Also #133 & #130

**Comment by msimberg** [comment]
> split should not try to detect multi-cast. It is precisely because of side-effects like printf() that this matters. split should always have the semantics of "replay" - only run the source once and store the result to share.

> Also #133 & #130

Thanks @kirkshoop for linking those, I missed them. As far as I'm concerned this issue is a duplicate of #130. `split` connecting and starting the given sender only once (at most once?) is exactly what I'd wish for it to do.
